onos-classic/templates/statefulset.yaml

```diff
@@ -0,0 +1,129 @@
+# This template provides a StatefulSet for ONOS nodes.
+# The StatefulSet is used to ensure that nodes are assigned a persistent identity.
+# The StatefulSet must be used with a headless service to ensure pods can communicate with each other through
+# their persistent identities.
+# The StatefulSet supports upgrades through a simple RollingUpdate strategy.
+# An anti-affinity policy is used to ensure pods are scheduled on distinct hosts when enabled.
+# To configure the pods, an init container is used to populate a properties file on an ephemeral volume. The
+# configuration volume is shared with the primary container for configuration purposes.
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ template "fullname" . }}
+  labels:
+    app: {{ template "fullname" . }}
+    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    release: "{{ .Release.Name }}"
+    heritage: "{{ .Release.Service }}"
+spec:
+  serviceName: {{ template "fullname" . }}-hs
+  selector:
+    matchLabels:
+      app: {{ template "fullname" . }}
+  replicas: {{ .Values.replicas }}
+  updateStrategy:
+    type: RollingUpdate
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: {{ template "fullname" . }}
+    spec:
+      {{- if .Values.podAntiAffinity.enabled }}
+      affinity:
+        podAntiAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+          - labelSelector:
+              matchExpressions:
+              - key: app
+                operator: In
+                values:
+                - {{ template "fullname" . }}
+            topologyKey: kubernetes.io/hostname
+      {{- end }}
+      initContainers:
+        - name: {{ .Chart.Name }}-init
+          image: tutum/dnsutils:latest
+          env:
+            - name: ATOMIX_SERVICE
+              value: {{ template "atomix.fullname" . }}
+            - name: ATOMIX_REPLICAS
+              value: {{ .Values.atomix.replicas | quote }}
+          command:
+            - sh
+            - -c
+            - "/scripts/configure-onos.sh $ATOMIX_SERVICE $ATOMIX_REPLICAS > /config/cluster.json && touch /config/active"
+          volumeMounts:
+            - name: init-scripts
+              mountPath: /scripts
+            - name: config
+              mountPath: /config
+      containers:
+      - name: {{ .Chart.Name }}
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        resources:
+          requests:
+            memory: {{ .Values.resources.requests.memory }}
+            cpu: {{ .Values.resources.requests.cpu }}
+        env:
+          - name: JAVA_OPTS
+            value: -Xmx{{ default "2G" .Values.heap }}
```

Comment:
This is set to `4G` in the values.yaml.  Is the `default` needed? 

How does this relate to `resources.requests.memory` ?

Reply:
Removed the default; it's ok for the heap to grow larger than the requested memory.